### PR TITLE
disable debug library postfix

### PR DESCRIPTION
**Pull request by dzosz**
DataDirLocater depeneds on specific library names and with custom postfix it failed to find libsync.so (renamed to libsyncd.so) in current directory. bug prevents spring to start/recognize correct datadir. affects Debug builds only.
 
https://github.com/beyond-all-reason/spring/blob/cbeb010dca48d56f9ce6d474e309992d671db7bb/rts/System/FileSystem/DataDirLocater.cpp#L61

commented out entire section as assimp shouldn't change our global project settings.

### CODE REVIEW

**Comment by p2004a** [comment]
Can you add a minimal repro step? I would like to test this before merging.

**Comment by dzosz** [comment]
> Can you add a minimal repro step? I would like to test this before merging.

I'm working with engine straight from dev environment without installing.
1. build debug version with gcc
2. create springsettings.cfg in local bld directory with variable `springData = /home/tommy/dev/bar-spring/bld-debug-gcc`
3. ./spring

**Review by p2004a — APPROVED** [review]
Sorry for the delay :frowning_face: !

**TLDR**: LGTM

Your reproduction steps describe what you do, but not the expected behavior and actual behavior. So, e.g. "place file here and it's not loaded". The part I've reproduced is that indeed because of this code, the spring enters the "install mode" and there isn't line with "Portable Mode!" in the logs. I'm not that familiar with that part and what the impact is exactly.

I myself always start spring with `--isolation` so it never tries to load any stuff from e.g. `~/.config/spring`.

What I've noticed is that `CMAKE_BUILD_TYPE` is case-sensitive (such an easy way to shut yourself in the foot), and almost everywhere I see we use fully uppercase. That's also maybe why we've not run into this one before: using `DEBUG` instead of `Debug` doesn't trigger this code :tired_face:

Anyway this change itself fully looks good to me, feel free to merge it. The only change I would recommend doing is just drop the code entirely, not just comment out (it's not working in default upper case setting, and dead code in repo is not useful).

**Comment by dzosz** [comment]
I'll look into possibility of adding debug logs for spring directory detection.
